pennylane_sf/remote.py

```diff
@@ -0,0 +1,172 @@
+                                  Interferometer, Ket, Kgate, MeasureFock,
+                                  Pgate, Rgate, S2gate, Sgate, Squeezed,
+                                  Thermal, Vgate)
+from strawberryfields.utils.post_processing import (all_fock_probs_pnr,
+                                                    samples_expectation,
+                                                    samples_variance)
+from strawberryfields.circuitspecs import circuit_db
+
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsRemote(StrawberryFieldsSimulator):
+    r"""StrawberryFields remote device for PennyLane.
+
+    A valid Strawberry Fields API token is needed for access. This token can be
+    passed when creating the device. The default configuration options of
+    Strawberry Fields are used to store the authentication token in a
+    configuration file.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+        shots (int): number of circuit evaluations/random samples used to
+            estimate expectation values of observables
+        backend (str): name of the remote backend to be used
+        hbar (float): the convention chosen in the canonical commutation
+            relation :math:`[x, p] = i \hbar`
+        sf_token (str): the SF API token used for remote access
+    """
+    name = "Strawberry Fields Hardware PennyLane plugin"
+    short_name = "strawberryfields.remote"
+
+    _operation_map = {
+        'CatState': Catstate,
+        'CoherentState': Coherent,
+        'FockDensityMatrix': DensityMatrix,
+        'DisplacedSqueezedState': DisplacedSqueezed,
+        'FockState': Fock,
+        'FockStateVector': Ket,
+        'SqueezedState': Squeezed,
+        'ThermalState': Thermal,
+        'GaussianState': Gaussian,
+        'Beamsplitter': BSgate,
+        'CrossKerr': CKgate,
+        'ControlledAddition': CXgate,
+        'ControlledPhase': CZgate,
+        'Displacement': Dgate,
+        'Kerr': Kgate,
+        'QuadraticPhase': Pgate,
+        'Rotation': Rgate,
+        'TwoModeSqueezing': S2gate,
+        'Squeezing': Sgate,
+        'CubicPhase': Vgate,
+        'Interferometer': Interferometer
+    }
+
+    _observable_map = {
+        "Identity": None,
+        "NumberOperator": None,
+        "TensorN": None,
+    }
+
+    def __init__(self, *, backend, shots=1000, hbar=2, sf_token=None):
+
+
+        # Explicitly obtain the default target if a family of backends was
+        # defined
+        if backend in sf.RemoteEngine.DEFAULT_TARGETS:
+            backend = sf.RemoteEngine.DEFAULT_TARGETS[backend]
+
+        # Infer the number of modes from the circuitspecs from Strawberry
+        # Fields
+        wires = circuit_db[backend].modes
+        super().__init__(wires, analytic=False, shots=shots, hbar=hbar)
+        self.backend = backend
+
+        if sf_token is not None:
+            sf.store_account(sf_token)
+
+    def reset(self):
```

Comment:
Added a `reset` method, as the inherited `StrawberryFieldsRemote.reset` would fail on `self.eng.reset()` when the quantum function is called multiple times.